pub-critic/src/app/components/ReviewForm/ReviewForm.tsx

```diff
@@ -0,0 +1,168 @@
+  gameId,
+  gameName,
+  initReview,
+  refetch,
+  isEdit,
+}: ReviewFormProps) => {
+  const [rating, setRating] = useState<number>(
+    initReview ? initReview.score : 0
+  );
+  const [title, setTitle] = useState<string>(
+    initReview ? initReview.title : ""
+  );
+  const [body, setBody] = useState<string>(initReview ? initReview.body : "");
+  const [error, setError] = useState<string>("");
+
+  const handleSubmit = async (e: React.FormEvent<HTMLFormElement>) => {
+    e.preventDefault();
+
+    console.log(rating, title, body);
+    console.log("submitted");
+
+    if (rating === 0) {
+      setError("Please select a rating");
+      return;
+    }
+
+    if (title.length < 3) {
+      setError("Title must be at least 3 characters long");
+      return;
+    }
+
+    if (body.length < 10) {
+      setError("Body must be at least 10 characters long");
+      return;
+    }
+
+    if (isEdit && body === initReview!.body && title === initReview!.title) {
+      setError("You must change something to update your review");
+      return;
+    }
+
+    const response = isEdit
+      ? await updateReview(initReview!.id, {
+          body,
+          title,
+          score: rating,
+          gameName,
+        })
+      : await postReview(
+          {
+            body,
+            title,
+            score: rating,
+            gameName,
+          },
+          gameId
+        );
+    if (response) {
+      alert(
+        isEdit ? "Review updated successfully" : "Review posted successfully"
+      );
+      refetch!();
+      return;
+    }
+    setError("Something went wrong");
+  };
+
+  return (
+    <form id="#review" onSubmit={handleSubmit} className={classes.form}>
+      <h2 className={classes.heading}>
+        {isEdit ? "Edit your review" : "Write a review"}
+      </h2>
+      <div className={classes.stars}>
+        <span>Rating: </span>
+        <label htmlFor="rating-1" onClick={() => setRating(1)}>
+          <Image
+            src={rating >= 1 ? star : lackOfStar}
+            alt="star"
+            layout="fill"
+          />
+        </label>
+        <label htmlFor="rating-2" onClick={() => setRating(2)}>
+          <Image
+            src={rating >= 2 ? star : lackOfStar}
+            alt="star"
+            layout="fill"
+          />
+        </label>
+        <label htmlFor="rating-3" onClick={() => setRating(3)}>
+          <Image
+            src={rating >= 3 ? star : lackOfStar}
+            alt="star"
+            layout="fill"
+          />
+        </label>
+        <label htmlFor="rating-4" onClick={() => setRating(4)}>
+          <Image
+            src={rating >= 4 ? star : lackOfStar}
+            alt="star"
+            layout="fill"
+          />
+        </label>
+        <label htmlFor="rating-5" onClick={() => setRating(5)}>
+          <Image
+            src={rating >= 5 ? star : lackOfStar}
+            alt="star"
+            layout="fill"
+          />
+        </label>
```

Comment:
1) What if you had hundreds or thousands of stars? 
2) avoid arrow expressions inside the render method (read more about performance issues). Instead, pass a method reference

Reply:
:white_check_mark: